build_helpers/buildNPMInternals.sh

```diff
@@ -5,15 +5,15 @@
 var glob = require('glob');
 var path = require('path');
 var fs = require('fs');
-var babel = require('babel-core');
+var babel = require('@babel/core');
 
 var internalPath = path.join(__dirname, '../internal');
 if (!fs.existsSync(internalPath)) {
   fs.mkdirSync(internalPath);
 }
 
 var providesModuleRegex = /@providesModule ([^\s*]+)/;
-var moduleRequireRegex = /=\s+require\((?:'|")([\w\.\/]+)(?:'|")\);/gm;
+var moduleRequireRegex = /require\((?:'|")([\w\.\/]+)(?:'|")\)/gm;
```

Comment:
In the entire codebase, FDT uses imports without `./`
so imports are written like `require("FixedDataTable")`

now I'm not sure why but during internal files build, we need them in the form `require("./FixedDataTable")`
(notice the `./`)

so the build script (this file) reads through the file contents after it's transpiled by babel, and then searches for the regex matching strings like `"= require("FixedDataTable")"` (notice the `"= "`)
Then it replaces the match by prepending a "`./"` before the path, so that it becomes `= require("./FixedDataTable")`

Now the problem was that, after the babel upgrade, this transformation wasn't seen. After digging through a lot, and blaming Babel, flow plugins, webpack, etc, I found out that it was simply because the requires were transpiled a bit differently:
```
var _FixedDataTableRow = _interopRequireDefault(require('FixedDataTableRow'));
```
instead of the earlier
```
var _FixedDataTableRow = require('./FixedDataTableRow');
```

notice that `"= require"` isn't present, and hence the regex didn't match...

After changing the regex, and the replace string down below, everything seems to work 🙂

Reply:
This is a pain, could we clean up our imports to be more idiomatic and remove this in the future?

Alternatively you might be able to use webpack's resolve.modules to resolve the imports from the root path rather than relatively which is what we do in LD.  It would still require specifying the path from the root rather than just the filename, but may be easier to migrate to than specifying relative path.

https://webpack.js.org/configuration/resolve#resolvemodules
https://opengrok.schrodinger.com/xref/seurat/SEURAT-JavaScript/webpack/common.config.js#68

Reply:
yep that makes sense to me

Reply:
to be clear, will the new PR will be for adding ./ in all the imports?

Reply:
Yea, I think the intent is to just make sure we won't need a build helper for converting the paths.
It can probably be avoided by adding "./" to all imports, but I don't know.

The build system in FDT puts everything into the Internal folder, which doesn't have any structure/directories. The entire thing is flat.

![image](https://user-images.githubusercontent.com/41563608/58195671-26653500-7ce6-11e9-8e54-13b2032fae7d.png)

That's probably why the simple "./[filename" works.